### PR TITLE
72 local verification for deck

### LINKED ISSUE #72: Local verification for deck
need to check the local deck dignity

### PR DESCRIPTION
**Pull request by whatdoes3plus1equalsto**
closes #72

### CODE REVIEW

**Review by kivine99 — COMMENTED** [review]
I think it might make sense for the deck manager to have a static instance of DeckValidator

**Comment by VanJackal** [comment]
> I think it might make sense for the deck manager to have a static instance of DeckValidator

maybe not static but as a public field would make sense

File: src/objects/DeckValidator.ts
Comment (VanJackal) [inline src/objects/DeckValidator.ts:1]: code here is fine, Im not sure if this is the architecture we want for this though.

in particular would we rather have the validator hold an instance of Deck and only operate on that instance, and the deck would have a reference to its validator.

that way you could call Deck.validator.getValidity (or just Deck.getValidity if we dont want to expose DeckValidtor)

Comment (kivine99) [reply src/objects/DeckValidator.ts:1]: The problem is that I have some weird things going on in the frontend that I would have to change if we had it like you're suggesting (and that could end up being a lot, I'm not the happiest with how I did it lol). Being able to call deckValidator.validate(cards[]) works perfectly for how I have it setup rn

Comment (VanJackal) [reply src/objects/DeckValidator.ts:1]: Would a validator under Client.User.Decks.validator work then?

Comment (kivine99) [reply src/objects/DeckValidator.ts:1]: I might've misunderstood your first comment

Comment (VanJackal) [reply src/objects/DeckValidator.ts:1]: Ok i think it should move from client to there then

File: src/IClient.ts [hunk 1]
```diff
@@ -11,6 +11,7 @@ export interface IClient {
     vendors:IVendorManager;
     offers: IOffersManager;
     rest:IRest;
+    deckValidator: IDeckValidator;
```
Comment (kivine99) [inline src/IClient.ts:14]: I don't think deckValidator needs to be in the client if it's in DeckManager

**Review by kivine99 — APPROVED** [review]

**Pull request merged**